larray/inout/common.py

```diff
@@ -1,11 +1,20 @@
 from __future__ import absolute_import, print_function
 
 import os
+from datetime import date, time, datetime
 from collections import OrderedDict
 
+from larray.util.compat import basestring
 from larray.core.array import Array
 
 
+# only for HDF5 and pickle formats
+# XXX: also include scalars in Excel and CSV formats (via __scalars__ special sheet/.csv file)?
+#      But then why saving scalars but not Axis and Group objects. That would be inconsistent.
+# XXX: support list and dict?
```

Comment:
@gdementen the two questions above are still open. What do you think?

Reply:
Q1: Even though I can see some value in the ability to save/dump scalars to Excel/CSV files even if groups/axes are not (even if that would be odd indeed), because that could be used to make some kind of control panel in an excel file to run a model, which would make for easy "GUI" for some models. **BUT** doing this via a Session might not be the best API (or at least this needs to be thought through) and in either case, IMO it does not need to be linked to this PR as I consider that lower priority.
Q2: That would be nice too, but also something with much lower priority IMO.

Reply:
Q2: list and tuple objects are already supported with the current implementation of `PandasHDFHandler` and `PickleHandler`. But in that case, ``__scalars__`` is no longer a valid name...

Reply:
??? They might be nearly supported but not entirely AFAICT as they do not satisfy any of the isinstance checks _dump_item, or am I missing something? Adding support for that might not be hard but requires additional tests, etc... In short it would required more time, so I would rule that as out of scope for now, but that's your call.

Reply:
For the lists and tuples, I just have to add `list, tuple` to the `_supported_scalar_types` protected variable. It works (I already added the tests in test_session.py). But in that case, using `__scalars__` is wrong. What do you think about `__non_larray__`?

Reply:
Now that I think of it, having groups in `__groups__` and axes in `__axes__` for HDF files does not seem to be useful (since we are guaranteed not to have name collisions -- because they come from the same session -- and we store the object type in node attributes anyway), or is it? So I would either store everything as leafs on the root node (like arrays) (*), or create a group node per "broad" type: __axes__, __groups__, __scalars__, __sequences__, __mappings__, ...

(*) the inevitable HDF file format change due to #761 might be the opportunity to make this change too.

Reply:
I would go for the first option (i.e. store everything as leafs on the root node).
Now, I'm thinking of it, we could use HDF groups to store several Session objects in the same HDF file?

Reply:
That's certainly possible and could be nifty, but I would like to avoid yet another feature nobody uses, and I don't see a practical use case for our current users yet, so **unless you see one**, I would wait until somebody asks for it (or until we find some real case where it would actually make things more practical).

Reply:
As a side note, moving everything to the root node has the disadvantage that we have to support two distinct file formats when reading HDF files (but since we will need that for #761 anyway, this is the right time to do it).

Reply:
About `list`, `tuple` and `dict`, I understand your remark so I will only implement save/load of scalars.